File: RealEstateApp.Domain/Entities/Property.cs
namespace RealEstateApp.Domain.Entities
{
    public class Property : IEntity
    {
        public Property()
        {
        }

        public string Id { get; set; } = default!;
        public required string Name { get; init; }
        public required string Address { get; init; }
        public decimal Price { get; init; }
        public required string CodeInternal { get; init; }
        public int Year { get; init; }

        // Foreign key for Owner
        public string IdOwner { get; init; }

        // Navigation properties
        public virtual Owner Owner { get; set; } = null!;
        public virtual ICollection<PropertyImage> PropertyImages { get; set; } = [];
        public virtual ICollection<PropertyTrace> PropertyTraces { get; set; } = [];
    }
}

File: RealEstateApp.ReactApp/src/models/Property.ts
/**
 * Represents a property owner
 */
export interface Owner {
  id: number;
  name: string;
  address: string;
  photo: string;
  birthday: string; // ISO date string
}

/**
 * Represents a property image
 */
export interface PropertyImage {
  id: number;
  file: string; // URL or file path
  enabled: boolean;
}

/**
 * Represents a real estate property
 */
export interface Property {
  id: string;
  name: string;
  address: string;
  price: number;
  codeInternal: string;
  year: number;
  idOwner: number;
  owner?: Owner;
  images: PropertyImage[];
}

/**
 * Represents a paginated list of properties
 */
export interface PropertyList {
  items: Property[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
}

/**
 * Represents filter criteria for property searches
 */
export interface PropertyFilter {
  name?: string;
  address?: string;
  minPrice?: number;
  maxPrice?: number;
  pageNumber?: number;
  pageSize?: number;
}